pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -0,0 +1,328 @@
+	IANAReverseInfoElements = []string{
+		"reverse_PacketTotalCount",
+		"reverse_OctetTotalCount",
+		"reverse_PacketDeltaCount",
+		"reverse_OctetDeltaCount",
+	}
+	AntreaInfoElements = []string{
+		"sourcePodName",
+		"sourcePodNamespace",
+		"sourceNodeName",
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+	}
+)
+
+type flowExporter struct {
+	flowRecords     *flowrecords.FlowRecords
+	process         ipfix.IPFIXExportingProcess
+	elementsList    []*ipfixentities.InfoElement
+	exportFrequency uint
+	pollInterval    time.Duration
+	templateID      uint16
+}
+
+func genObservationID() (uint32, error) {
+	name, err := env.GetNodeName()
+	if err != nil {
+		return 0, err
+	}
+	h := fnv.New32()
+	h.Write([]byte(name))
+	return h.Sum32(), nil
+}
+
+func NewFlowExporter(records *flowrecords.FlowRecords, expProcess ipfix.IPFIXExportingProcess, elemList []*ipfixentities.InfoElement, exportFrequency uint, pollInterval time.Duration, tempID uint16) *flowExporter {
+	return &flowExporter{
+		records,
+		expProcess,
+		elemList,
+		exportFrequency,
+		pollInterval,
+		tempID,
+	}
+}
+
+func InitFlowExporter(collector net.Addr, records *flowrecords.FlowRecords, exportFrequency uint, pollInterval time.Duration) (*flowExporter, error) {
+	// Create IPFIX exporting expProcess and initialize registries and other related entities
+	obsID, err := genObservationID()
+	if err != nil {
+		return nil, fmt.Errorf("cannot generate obsID for IPFIX ipfixexport: %v", err)
+	}
+
+	var expProcess ipfix.IPFIXExportingProcess
+	if collector.Network() == "tcp" {
+		// TCP transport do not need any tempRefTimeout, so sending 0.
+		expProcess, err = ipfix.NewIPFIXExportingProcess(collector, obsID, 0)
+	} else {
+		// For UDP transport, hardcoding tempRefTimeout value as 1800s.
+		expProcess, err = ipfix.NewIPFIXExportingProcess(collector, obsID, 1800)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error while initializing IPFIX exporting expProcess: %v", err)
+	}
+	expProcess.LoadRegistries()
+
+	flowExp := NewFlowExporter(records, expProcess, nil, exportFrequency, pollInterval, expProcess.NewTemplateID())
+
+	templateRec := ipfix.NewIPFIXTemplateRecord(uint16(len(IANAInfoElements)+len(IANAReverseInfoElements)+len(AntreaInfoElements)), flowExp.templateID)
+
+	sentBytes, err := flowExp.sendTemplateRecord(templateRec)
+	if err != nil {
+		return nil, fmt.Errorf("error while creating and sending template record through IPFIX process: %v", err)
+	}
+	klog.V(2).Infof("Initialized flow exporter and sent %d bytes size of template record", sentBytes)
+
+	return flowExp, nil
+}
+
+// Run enables to export flow records periodically at a given flow export frequency
+func (exp *flowExporter) Run(stopCh <-chan struct{}, pollDone <-chan struct{}) {
+	klog.Infof("Start exporting IPFIX flow records")
+	ticker := time.NewTicker(time.Duration(exp.exportFrequency) * exp.pollInterval)
```

Comment:
do you think we need the ticker here instead of just counting the number of messages received on `pollDone`?